src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,60 @@
  */
 
 #include <app/clusters/scenes-server/SceneHandlerImpl.h>
+#include <app/util/endpoint-config-api.h>
 
 namespace chip {
 namespace scenes {
 
+namespace {
+
+using ConcreteAttributePath  = app::ConcreteAttributePath;
+using AttributeValuePairType = app::Clusters::ScenesManagement::Structs::AttributeValuePair::Type;
+
+/// CapAttributeID
+/// Cap the attribute value based on the attribute type size
+/// @param[in] aVPair   AttributeValuePairType
+/// @param[in] metadata  EmberAfAttributeMetadata
+///
+void CapAttributeID(AttributeValuePairType & aVPair, const EmberAfAttributeMetadata * metadata)
+{
+    // Calculate the maximum value that can be represented with the given number of bytes
+    uint64_t maxValue = (1ULL << (metadata->size * 8)) - 1;
```

Comment:
should we consider unsigned types? That starts being difficult.

What sizes do we have for arrays or strings ? do we assume this is never called for such a thing (makes sense, however maybe we should check)

Reply:
also BOOLEAN will have size 1, however cap is probably 1.

Reply:
The only acceptable types are signed and unsigned integers from  8bits to 64bits. I am not sure how we could consider signed types here.